src/main/java/org/wildfly/naming/client/ProviderEnvironment.java

```diff
@@ -293,52 +290,21 @@ public Builder populateFromEnvironment(Map<String, ?> environment) throws Naming
             final String securityProtocol = getEnvString(environment, Context.SECURITY_PROTOCOL, null, true);
             final String globalSslEnabledOption = getEnvString(environment, EJB_REMOTE_CONNECTION_PROVIDER_PREFIX + Options.SSL_ENABLED, null, true);
             final boolean isSsl = globalSslEnabledOption != null ? Boolean.parseBoolean(globalSslEnabledOption) : securityProtocol != null && "ssl".equalsIgnoreCase(securityProtocol.trim());
-            final String callbackClass = getEnvString(environment, CALLBACK_HANDLER_KEY, null, true);
             final IdentityCredentials credentials = EnvironmentUtils.getSecurityCredentials(environment);
-            final String passwordBase64 = getEnvString(environment, PASSWORD_BASE64_KEY, null, false);
-            final String securityRealm = getEnvString(environment, REALM_KEY, null, true);
 
-            if (callbackClass != null && (userName != null || credentials != null || passwordBase64 != null)) {
-                throw Messages.log.callbackHandlerAndUsernameAndPasswordSpecified();
-            }
 
             // we definitely must override default auth if any of these are given; we _may_ have to do so if compat props are given
-            boolean overrideDefaultAuth = credentials != null || passwordBase64 != null || callbackClass != null || securityRealm != null || userName != null;
-            CallbackHandler callbackHandler = null;
-
-            if (callbackClass != null) {
-                try {
-                    final Class<?> clazz = Class.forName(callbackClass, true, classLoader);
-                    callbackHandler = (CallbackHandler) clazz.newInstance();
-                } catch (ClassNotFoundException e) {
-                    throw Messages.log.failedToLoadCallbackHandlerClass(e, callbackClass);
-                } catch (Exception e) {
-                    throw Messages.log.failedToInstantiateCallbackHandlerInstance(e, callbackClass);
-                }
-            }
+            boolean overrideDefaultAuth = credentials != null || userName != null;
 
             OptionMap remotingOptions = getOptionMap(environment, CONNECT_OPTIONS_PREFIX, Builder.class.getClassLoader());
 
-            if (callbackHandler != null || userName != null) {
```

Comment:
Only _callbackHandler != null_ check should be removed from this if condition and not whole _if_ block IMHO.

Reply:
Do we have a test covering this? If now, perhaps we should.

Reply:
I have fixed the mentioned issue and added the test that checks whether global env principal and credentials are propagated to the AuthenticationConfiguration.